tests/components/insteon/test_init.py

```diff
@@ -0,0 +1,227 @@
+
+from homeassistant.components import insteon
+from homeassistant.components.insteon.const import (
+    CONF_CAT,
+    CONF_OVERRIDE,
+    CONF_SUBCAT,
+    CONF_X10,
+    DOMAIN,
+    PORT_HUB_V1,
+    PORT_HUB_V2,
+)
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICE,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.setup import async_setup_component
+
+from .const import (
+    MOCK_ADDRESS,
+    MOCK_CAT,
+    MOCK_IMPORT_CONFIG_PLM,
+    MOCK_IMPORT_FULL_CONFIG_HUB_V1,
+    MOCK_IMPORT_FULL_CONFIG_HUB_V2,
+    MOCK_IMPORT_FULL_CONFIG_PLM,
+    MOCK_IMPORT_MINIMUM_HUB_V1,
+    MOCK_IMPORT_MINIMUM_HUB_V2,
+    MOCK_SUBCAT,
+    MOCK_USER_INPUT_PLM,
+    PATCH_CONNECTION,
+)
+from .mock_devices import MockDevices
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def mock_successful_connection(*args, **kwargs):
+    """Return a successful connection."""
+    return True
+
+
+async def mock_failed_connection(*args, **kwargs):
+    """Return a failed connection."""
+    raise ConnectionError("Connection failed")
+
+
+async def test_setup_entry(hass: HomeAssistantType):
+    """Test setting up the entry."""
+    config_entry = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_INPUT_PLM)
+    config_entry.add_to_hass(hass)
+
+    with patch.object(
+        insteon, "async_connect", new=mock_successful_connection
+    ), patch.object(insteon, "async_close") as mock_close, patch.object(
+        insteon, "devices", new=MockDevices()
+    ):
+        assert await async_setup_component(
+            hass,
+            insteon.DOMAIN,
+            {},
+        )
+        await hass.async_block_till_done()
+        hass.bus.async_fire(EVENT_HOMEASSISTANT_STOP)
+        await hass.async_block_till_done()
+        # pylint: disable=no-member
+        assert insteon.devices.async_save.call_count == 1
+        assert mock_close.called
+
+
+async def test_import_plm(hass: HomeAssistantType):
+    """Test setting up the entry from YAML to a PLM."""
+    config = {}
+    config[DOMAIN] = MOCK_IMPORT_CONFIG_PLM
+
+    with patch.object(
+        insteon, "async_connect", new=mock_successful_connection
+    ), patch.object(insteon, "close_insteon_connection"), patch.object(
+        insteon, "devices", new=MockDevices()
+    ), patch(
+        PATCH_CONNECTION, new=mock_successful_connection
+    ):
+        assert await async_setup_component(
+            hass,
+            insteon.DOMAIN,
+            config,
+        )
+        await hass.async_block_till_done()
+        await asyncio.sleep(0.01)
+    assert hass.config_entries.async_entries(DOMAIN)
+    data = hass.config_entries.async_entries(DOMAIN)[0].data
+    assert data[CONF_DEVICE] == MOCK_IMPORT_CONFIG_PLM[CONF_PORT]
+    assert CONF_PORT not in data
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)